engine/minskyTensorOps.cc

```diff
@@ -368,17 +368,23 @@ namespace minsky
   {
     std::shared_ptr<ITensor> arg1, arg2;
     void computeTensor() const override {//TODO Sparse implementation
-      size_t m=arg1->hypercube().numElements(), n=arg2->hypercube().numElements();   
+      size_t m=arg1->size(), n=arg2->size();   
   	
-      for (size_t i=0; i< m; i++)
-      {
-        auto v1=(*arg1)[i];  			
-        for (size_t j=0; j< n; j++) 
-        {
-            auto v2=(*arg2)[j];			
-            if (!isnan(v1) && !isnan(v2)) cachedResult[i+j*m]=v1*v2;			
-		}
-	  }
+      //if (cachedResult.index().empty())  	
+       for (size_t i=0; i< m; i++)
+       {
+         auto v1=(*arg1)[i];  			
+         for (size_t j=0; j< n; j++) 
+         {
+             auto v2=(*arg2)[j];			
+             if (!isnan(v1) && !isnan(v2)) cachedResult[i+j*m]=v1*v2;			
```

Comment:
This should be fine. Note the test for isnan is not necessary, a nan multiplied by anything else is still a nan.

Reply:
Ok, I got rid of the conditional.